src/Lachain.Storage/Repositories/LocalTransactionRepository.cs

```diff
@@ -33,32 +33,47 @@ public void TryAddTransaction(TransactionReceipt receipt)
         {
             try
             {
-                string[] signatures = {
-                    "transfer(address,uint256)", 
+                string[] signatures =
+                {
+                    "transfer(address,uint256)",
                     "transferFrom(address,address,uint256)",
                     "mint(address,uint256)",
                     "burn(address,uint256)"
                 };
-                
+
                 var decoder = new ContractDecoderLtr(receipt.Transaction.Invocation.ToArray());
-                object[] decodedRes = {};
-                
+                object[] decodedRes = { };
+
                 foreach (var signature in signatures)
                 {
                     decodedRes = decoder.Decode(signature);
                     break;
                 }
-                
-                if (decodedRes[0] is UInt160 decodedAddr && decodedAddr.Equals(receipt.Transaction.To))
+
+                if (decodedRes.Length == 3)
                 {
-                    var temp = LoadState();
-                    SaveState(temp.Concat(receipt.Hash.ToBytes()).ToArray());     
+                    if (decodedRes[0] is UInt160 fromAddr && fromAddr.Equals(receipt.Transaction.From) &&
```

Comment:
1)this checks should be inside the cycle with signatures
2)why do you compare receipt.Transaction params with decoded script call arguments? Your goal is to fiind local address among this arguments,  transaction params are not related to this search